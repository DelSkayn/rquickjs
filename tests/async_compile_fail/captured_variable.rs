use rquickjs::{async_with, AsyncContext, AsyncRuntime};

pub async fn test() {
    let rt = AsyncRuntime::new().unwrap();
    let ctx = AsyncContext::full(&rt).await.unwrap();

    let fut = {
        let mut var = 1u32;
        let var_ref = &mut var;
        async_with!(ctx => |_ctx|{
            *var_ref += 1;
        })
    };
    fut.await;
}

fn main() {}
