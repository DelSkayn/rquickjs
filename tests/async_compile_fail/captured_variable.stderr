error[E0597]: `var` does not live long enough
  --> tests/async_compile_fail/captured_variable.rs:9:23
   |
7  |     let fut = {
   |         --- borrow later stored here
8  |         let mut var = 1u32;
   |             ------- binding `var` declared here
9  |         let var_ref = &mut var;
   |                       ^^^^^^^^ borrowed value does not live long enough
...
13 |     };
   |     - `var` dropped here while still borrowed
